src/seedu/addressbook/data/AddressBook.java

```diff
@@ -44,6 +48,8 @@ public AddressBook(UniquePersonList persons, UniqueTagList tags) {
         for (Person p : allPersons) {
             syncTagsWithMasterList(p);
         }
+        taggings = new ArrayList<Tagging>();
+        
```

Comment:
you seem to have missed the next part of the exercise where you need to print the tags added/deleted on exit